src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoData.cs

```diff
@@ -93,5 +97,22 @@ internal static int[] GetCurrencyNegativePatterns(CultureInfo cultureInfo)
 
             throw DateTimeFormatInfoData.GetCultureNotSupportedException(cultureInfo);
         }
+        
+        private static Version GetOSXKernelVersion()
+        {
+            if (s_isOSX)
+            {
+                ulong bytesLength = 256;
+                byte[] bytes = new byte[bytesLength];
+                sysctlbyname("kern.osrelease", bytes, ref bytesLength, null, 0);
+                string versionString = Encoding.UTF8.GetString(bytes);
+                return Version.Parse(versionString);
+            }
+            
+            return new Version(0, 0, 0);
+        }
+        
+        [DllImport("libc")]
+        private static extern int sysctlbyname(string ctlName, byte[] oldp, ref ulong oldpLen, byte[] newp, ulong newpLen);
```

Comment:
> the test should read the OS data and validate it against what the framework is providing.

FWIW, I'm not convinced that's much better.  Then we're essentially just writing the same code twice and validating that we got the same answer in both cases, in the code in the product that reads it and in the code in the test that reads it.  While it would help to catch bugs due to typos and the like, if we misunderstand the structure of the data or how it's meant to be interpreted, we're likely to make that mistake in both places.

This is also just one example of a much larger issue: the underlying OS adds features or makes changes, there's no good way in managed code to do feature detection, and the underlying systems actually advertise that those features are available starting in a particular version, yet we don't expose such version information in a consumable fashion and instead state that developers shouldn't use version numbers for this purpose.  For example, the man page for getrandom (http://man7.org/linux/man-pages/man2/getrandom.2.html) states it "was introduced in version 3.17 of the Linux kernel"... so what's our recommendation for that case?